File: C++/Sort/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    vector<int> temp;
    int idx, i, j, k;

    for (idx = 0; idx < commands.size(); idx++) {
        i = commands[idx][0] - 1;
        j = commands[idx][1] - 1;
        k = commands[idx][2] - 1;
        
        for (; i <= j; i++)
            temp.push_back(array[i]);

        sort(temp.begin(), temp.end());
        answer.push_back(temp[k]);

        temp.clear();
    }

    return answer;
}

File: Python/Sort/k번째수.py
def solution(array, commands):
    answer = []

    for cmd in commands:
        i, j, k = cmd[0], cmd[1], cmd[2]
        sub = array[i-1:j]
        sub.sort()
        answer.append(sub[k-1])

    return answer

arr = [1, 5, 2, 6, 3, 7, 4]
cmd = [[2, 5, 3], [4, 4, 1], [1, 7, 3]]

print(solution(arr, cmd))